doitlive.py

```diff
@@ -181,6 +184,18 @@ class TTY(object):
     DIM = ANSICode(dim=True)
 
 
+def echo(message=None, file=None, nl=True, err=False, color=None, carriage_return=False):
```

Comment:
defining echo like this will break full compability with clicks echo function though, because, you can no longer call:
```python
echo("test", None, False)
```
which is an ugly version of:
```python
echo("test", nl=False)
```
I'd like to retain this compability in the carriage return functionality gets merged into click.